hpedockerplugin/hpe_storage_api.py

```diff
@@ -463,11 +463,16 @@ def volumedriver_create(self, name, opts=None):
                     LOG.error(msg)
                     return json.dumps({u"Err": six.text_type(msg)})
 
+        # check for valid promoteSnap option and reurn the result
+        if ('Opts' in contents and contents['Opts'] and
+                'promote' in contents['Opts']):
+             return self.revert_to_snapshot(name, opts)
+
         # snapshotOf and cloneOf are mutually exclusive
         if ('Opts' in contents and contents['Opts'] and
                 'snapshotOf' in contents['Opts'] and
                 'cloneOf' in contents['Opts']):
-            msg = (_('both snapshoOf and cloneOf cannot be specified at the '
+            msg = (_('both snapshotOf and cloneOf cannot be specified at the '

```

Comment:
We should even add a check for 'promote' here as it cannot be clubbed with 'snapshotOf' or 'cloneOf'.

Reply:
OKay

Reply:
I will log  "while reverting volume to snapshot state only valid option is promote=<vol_name>"